src/vsr/replica.zig

```diff
@@ -8967,11 +8968,17 @@ const DVCQuorum = struct {
                 if (header_nacks.isSet(header_index)) {
                     // The op is nacked explicitly.
                     nacks += 1;
-                } else if (vsr.Headers.dvc_header_type(header) == .valid and
-                    header_canonical != null and header_canonical.?.checksum != header.checksum)
-                {
-                    // The op is nacked implicitly, because the replica has a different header.
-                    nacks += 1;
+                } else if (vsr.Headers.dvc_header_type(header) == .valid) {
+                    if (header_canonical != null and header_canonical.?.checksum != header.checksum) {
+                        // The op is nacked implicitly, because the replica has a different header.
+                        nacks += 1;
+                    }
+                    if (header_canonical == null and header.view <= log_view_canonical) {
```

Comment:
:thinking: Why `<= log_view_canonical` rather than `< log_view_canonical`?

First of all: Isn't `header.view <= log_view_canonical` _always_ true for `header` (even outside this branch)?
If `header.view > log_view_canonical`, then that means that `header` was prepared in a newer-than-canonical view, which means that this DVC's `log_view` should be greater than `log_view_canonical`, which is impossible.

(On that note, can we `assert(header.view <= log_view_canonical);` we define `const header =` above?)

And second: The conjunction `header_canonical == null and header.view == log_view_canonical` should be impossible, since `header.view == log_view_canonical` implies that this DVC's `log_view` is exactly `log_view_canonical`. But if this DVC's `log_view` is `log_view_canonical`, then this DVC is canonical, so `header_canonical == header`, and therefore `header_canonical != null`.

Reply:
Typo! Should definitely be just `<`!

Amusingly, it was `<` in the first version: 

https://github.com/tigerbeetle/tigerbeetle/pull/1517/files#diff-e95e4dc6c1aae349e798cfeed22ad01f830cce0e654e6b4b765b3e7b0c677b6aR8977

, but I forgot about that WIP PR when preparing this version.